### ISSUE TITLE
GPU device plugin interval health check

### ISSUE THREAD

**Issue by nadav213000** [NVIDIA/kubevirt-gpu-device-plugin#97]
Hey,

We use NVIDIA GPU Operator on OpenShift to expose Passthrough GPU with KubeVirt.

### Issue

We experienced an issue when one of the GPUs on the Node became unavailable, but the Node didn't change the reported GPU capacity or Alloctable resources. The GPU itself wasn't available and when I tried to create a new VM it reached `crashLoopBack` state until the GPU became available again.

Only after I restarted the pod `nvidia-sandbox-device-plugin-daemonset` on the specific Node the number of Alloctable and Capacity GPUs changed to the right number.

I checked the pods on this Nodes:
- `nvidia-sandbox-device-plugin`
- `nvidia-sandbox-validator`
- `nvidia-vfio-manager`
and there were no errors in their logs and I couldn't see any new logs from these pods. 

It looks like the pods run an initial healthCheck, and then don't run them again. Is there a way to make the Operator pods validate the health of the GPUs on an interval, so the resources available on the Node will be reflected correctly?

### How to reproduce

I reproduced the issue by logically removing one of the GPU PCI devices from the node using the command:

```bash
echo "1" > /sys/bus/pci/devices/<gpu_pci_id>/remove
```

and validated the GPU is no longer visible from the host using `lspci`.

Then, using `oc describe <node>` the number of GPUs exposed didn't change. After restarting the sandbox pod, the number of GPUs was updated to the right number.

To re-add the GPU you can run the command:
```bash
echo "1" > /sys/bus/pci/rescan
```
and restart the sandbox pod again

### Versions
- `NVIDIA Operator` - 23.6.0
- `OpenShift` - 4.12.35
- `nvidia sandbox device plugin` image - `nvcr.io/nvidia/cloud-native/gpu-operator-validator@sha256:b65eb649188193f39e169af5650acfc7aa3cc32d2328630118702f04fdc4afc1`

**Comment by doronkg** [comment]
Hey @rthallisey, I'm writing here on behalf of my colleague @nadav213000.
It seems as the resolution to this issue was brought in #105, and is introduced in v.1.2.8, correct?

**Comment by visheshtanksale** [comment]
> Hey @rthallisey, I'm writing here on behalf of my colleague @nadav213000. It seems as the resolution to this issue was brought in #105, and is introduced in v.1.2.8, correct?

Yes, https://github.com/NVIDIA/kubevirt-gpu-device-plugin/pull/105 resolves the scenarios that you have mentioned here and is released with v1.2.8.

**Comment by cdesiniotis** [comment]
@nadav213000 GPU Operator 24.6.0 has been released and contains the fix for this issue.
